### PR TITLE
Completed POST post 

### LINKED ISSUE #6: As an author I want to make public posts [BACKEND]
Creation URL ://service/authors/{AUTHOR_ID}/posts/
POST [local] create a new post but generate a new id

### PR DESCRIPTION
**Pull request by amandanguyeen**
Resolves #6, #19 
I think it resolves 19, it just stores the description as plain string
As an author, I want to make public posts.

Right now, ALL fields except title, description and author are nullable. I thought it would be easier to test on the front end but will eventually have to validate against that

I haven't written tests yet bc its easier to test first if my GET are working so i'll implement it then 

Post is missing comment information 

Screen Shot 2022-10-19 at 2 40 44 PM

### CODE REVIEW

File: mysocial/post/views.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+class CreationPostView(GenericAPIView):
+    serializer_class = CreatePostSerializer
+
+    def post(self, request, *args, **kwargs):
+        try:
+            serializer = self.serializer_class(data=request.data)
+            if serializer.is_valid():
+                data = serializer.data
+                data['author'] = Author.objects.get(official_id = kwargs['author_id'])
```
Comment (amandanguyeen) [inline mysocial/post/views.py:20]: @TurnipXenon kwargs is a lot easier to implement than writing down all the parameters in our url if you wanted to make the switch but i don't think itd matter too much

Comment (TurnipXenon) [reply mysocial/post/views.py:20]: Ahh! Nice tip! But I think with smaller, well-defined argument counts, I personally prefer explicitly declaring it and giving type hinting. There's less room for bugs plus I heavily rely on my IDE to spot errors. Although, I've not known this before. It seems more versatile with change, useful with tons of parameters, and maybe more pythonic though 👀 Thank you for the tip!

File: endpoints.txt [hunk 2]
```diff
@@ -0,0 +1,11 @@
+TO CREATE A POST:
+POST /authors/{AUTHOR_UUID}/posts/
```
Comment (TurnipXenon) [inline endpoints.txt:2]: Nice!!!! :P

**Comment by TurnipXenon** [comment]
Django migrate can't detect because of missing `__init__.py` T.T

**Review by TurnipXenon — APPROVED** [review]
I forgot! Yeah let's merge this! 🎉

**Pull request merged**